internal/boxcli/shellenv.go

```diff
@@ -85,6 +116,7 @@ func shellEnvFunc(
 	box, err := devbox.Open(&devopt.Opts{
 		Dir:               flags.config.path,
 		Environment:       flags.config.environment,
+		EnvForPackageBins: flags.envForPackageBins,
```

Comment:
As you already mentioned, I think it might be best to move this to each individual function that needs it.

Reply:
I'll give it a shot, but I fear it may add a lot of complexity/noise in the code.

Reply:
see #2159